delivery_carrier_label_gls/report/label.py

```diff
@@ -0,0 +1,418 @@
+    "weight": {'max_size': 5, 'required': True},
+}
+DELIVERY_MODEL = {
+    # 'address': ADDRESS_MODEL,
+    "consignee_ref":    {'max_size': 20},
+    "additional_ref_1": {'max_size': 20},
+    "additional_ref_2": {'max_size': 20},
+    "shipping_date":    {'date': '%Y%m%d', 'required': True},
+    "commentary":       {'max_size': 35},
+    "parcel_total_number": {'max_number': 999, 'type': int, 'required': True},
+    # TODO : valider 10 caract numériques
+}
+SENDER_MODEL = {
+    "customer_id":       {'max_size': 10, 'required': True},
+    "contact_id":        {'max_size': 10},
+    "outbound_depot":    {'max_size': 6, 'required': True},
+    "shipper_name":      {'max_size': 35, 'required': True},
+    "shipper_street":    {'max_size': 35, 'required': True},
+    "shipper_street2":   {'max_size': 35},
+    "shipper_zip":       {'max_size': 10, 'required': True},
+    "shipper_city":      {'max_size': 35, 'required': True},
+    "shipper_country":   {'in': GLS_COUNTRIES_PREFIX, 'required': True},
+}
+
+# Here is all fields called in mako template
+ADDRESS_MAPPING = {
+    'T860': "consignee_name",
+    'T8906': "contact",
+    'T863': "street",
+    'T861': "street2",
+    'T862': "street3",
+    'T330': "zip",
+    'T864': "city",
+    'T100': "country_code",
+    'T871': "consignee_phone",
+    'T1230': "consignee_mobile",
+    'T1229': "consignee_email",
+}
+PARCEL_MAPPING = {
+    'T530': "weight",
+    'T8973': "parcel_number_barcode",
+    'T8904': "parcel_number_label",
+}
+DELIVERY_MAPPING = {
+    # 'address': ADDRESS_MODEL,
+    'T859': "consignee_ref",
+    'T854': "additional_ref_1",
+    'T8907': "additional_ref_1",
+    'T8908': "additional_ref_2",
+    'T540': "shipping_date",
+    'T8318': "commentary",
+    'T8975': "gls_origin_reference",
+    'T8912': "gls_origin_reference",
+    'T8905': "parcel_total_number",  # TODO : valider 10 caract numériques
+    'T8702': "parcel_total_number",
+}
+ACCOUNT_MAPPING = {
+    'T8915': "customer_id",
+    'T805': "customer_id",
+    'T8914': "contact_id",
+    'T8700': "outbound_depot",
+    'T811': "shipper_street",
+    'T820': "shipper_street2",
+    'T810': "shipper_name",
+    'T822': "shipper_zip",
+    'T823': "shipper_city",
+    'T821': "shipper_country",
+}
+
+MAPPING = {}
+MAPPING.update(ACCOUNT_MAPPING)
+MAPPING.update(DELIVERY_MAPPING)
+MAPPING.update(PARCEL_MAPPING)
+MAPPING.update(ADDRESS_MAPPING)
+
+
+def dict_to_gls_data(params):
+    res = r'\\\\\GLS\\\\\|'
+    for key, val in params.items():
+        if val != '':
+            res += "%s:%s|" % (key, val)
+    res += r'/////GLS/////'
+    return res
+
+
+def gls_data_to_dict(data):
+    res = {}
+    for val in data.split('|')[1:-1]:
+        key, value = val.split(':', 1)
+        res[key] = value.decode(WEB_SERVICE_CODING, 'ignore')
+    return res
+
+
+def gls_decode(data):
+    return gls_data_to_dict(data)
+
+
+class Gls(AbstractLabel):
```

Comment:
GLS in capital letters

class name could be a bit more explicit. e.g. GLSLabel